db/migrate/20171116110243_migrate_existing_scenarios_for_public_transport_changes.rb

```diff
@@ -0,0 +1,98 @@
+  require 'progress_bar'
+
+  def up
+    # Old input key -> new input key
+    renamed = {
+      'transport_useful_demand_car_kms' => 'transport_useful_demand_passenger_kms',
+      'transport_useful_demand_truck_kms' => 'transport_useful_demand_freight_tonne_kms',
+      'transport_car_using_diesel_mix_efficiency' => 'transport_vehicle_combustion_engine_efficiency',
+      'transport_car_using_electricity_efficiency' => 'transport_vehicle_using_electricity_efficiency',
+      'transport_car_using_hydrogen_efficiency' => 'transport_vehicle_using_hydrogen_efficiency',
+      'transport_useful_demand_planes_efficiency' => 'transport_planes_efficiency',
+      'transport_useful_demand_ships_efficiency' => 'transport_ships_efficiency',
+      'transport_useful_demand_trains_efficiency' => 'transport_trains_efficiency',
+      'transport_train_using_coal_share' => 'transport_passenger_train_using_coal_share',
+      'transport_train_using_diesel_share' => 'transport_passenger_train_using_diesel_mix_share',
+      'transport_train_using_electricity_share' => 'transport_passenger_train_using_electricity_share',
+    }
+
+    # Assume that the share of 'trucks' are the same for 'busses'
+    # Assume that the share of 'trains passengers' are the same for 'trains freight'
+    assumptions = {
+      'transport_passenger_train_using_coal_share' => 'transport_freight_train_using_coal_share',
+      'transport_passenger_train_using_diesel_mix_share' => 'transport_freight_train_using_diesel_mix_share',
+      'transport_passenger_train_using_electricity_share' => 'transport_freight_train_using_electricity_share',
+      'transport_truck_using_compressed_natural_gas_share' => 'transport_bus_using_compressed_natural_gas_share',
+      'transport_truck_using_diesel_mix_share' => 'transport_bus_using_diesel_mix_share',
+      'transport_truck_using_electricity_share' => 'transport_bus_using_electricity_share',
+      'transport_truck_using_gasoline_mix_share' => 'transport_bus_using_gasoline_mix_share',
+      'transport_truck_using_hydrogen_share' => 'transport_bus_using_hydrogen_share',
+      'transport_truck_using_lpg_share' => 'transport_bus_using_lpg_share',
+      'transport_car_using_electricity_share' => 'transport_motorcycle_using_electricity_share'
+    }
+
+    corrections = {
+      'transport_motorcycle_using_electricity_share' => 'transport_motorcycle_using_gasoline_mix_share'
+    }
+
+    # Pruned input keys
+    removed = [
+      'transport_useful_demand_ship_kms',
+      'transport_useful_demand_trains',
+      'transport_useful_demand_planes',
+      'transport_truck_using_compressed_natural_gas_efficiency'
+    ]
+
+    scenarios = Scenario.where("`user_values` IS NOT NULL AND `balanced_values` IS NOT NULL AND `user_values` != '--- !ruby/hash:ActiveSupport::HashWithIndifferentAccess {}\n' AND `balanced_values` != '--- !ruby/hash:ActiveSupport::HashWithIndifferentAccess {}\n'")
```

Comment:
```sql
`user_values` != '--- !ruby/hash:ActiveSupport::HashWithIndifferentAccess {}\n' AND
`balanced_values` != '--- !ruby/hash:ActiveSupport::HashWithIndifferentAccess {}\n'")
```

Will this not miss scenarios where `user_values` has inputs and `balanced_values` is empty, and vice-versa?

---

```sql
SELECT count(id) FROM scenarios
-- 598742

SELECT count(id) FROM scenarios
WHERE
	`user_values` IS NOT NULL AND
	`balanced_values` IS NOT NULL AND
	`user_values` != '--- !ruby/hash:ActiveSupport::HashWithIndifferentAccess {}\n' AND
	`balanced_values` != '--- !ruby/hash:ActiveSupport::HashWithIndifferentAccess {}\n'
-- 77119

SELECT count(id) FROM scenarios
WHERE
	( `user_values` IS NOT NULL OR
		`balanced_values` IS NOT NULL )
	AND
	( `user_values` != '--- !ruby/hash:ActiveSupport::HashWithIndifferentAccess {}\n' OR
		`balanced_values` != '--- !ruby/hash:ActiveSupport::HashWithIndifferentAccess {}\n' )
-- 414564
```